CONTRIBUTING.md

```diff
@@ -63,13 +63,14 @@ We defined three categories of tests.
 - Integration test: Those tests cover the integration between the server and the dependencies (Cassandra, SQL, ES etc.).
 - Functional test: Those tests cover the E2E functionality of Temporal server. They are all under ./tests directory.
 
-Integration and functional tests require runtime dependencies. They can be run with `start-dependencies` target (uses `docker compose` internally). Open new terminal window and run:
+Integration and functional tests require runtime dependencies. If running unit tests, no need to start the dependencies. 
+They can be run with `start-dependencies` target (uses `docker compose` internally). Open new terminal window and run:
 
 ```bash
 make start-dependencies
 ```
 
-Before testing on macOS, make sure you increase the file handle limit:
+The default file handle limit on macOS should be sufficient to run the tests. If needed, increase the file handle limit:
```

Comment:
You can probably just remove this comment altogether, I've never seen us exceed this limit.